### ISSUE TITLE
Storybook v7 support?

### ISSUE THREAD

**Issue by mikestu** [nutboltu/storybook-addon-mock#148]
Storybok v7 is currently alpha, with beta targeted for this quarter (before end of year), and stable first quarter of next year.  We would like to begin using v7 alpha, as it has much better Vite support.  It works for us on all stories except the ones requiring this addon, and it appears that changes must be needed for it to support v7.  Any chance of putting out an alpha release of storybook-addon-mock, with v7 support?  If it works for others already, I must be doing something wrong, and can post more as a follow-up, or open a new issue.  Thanks!

**Comment by nutboltu** [comment]
I haven't tried v7. I might need sometime to investigate further on v7

**Comment by mikestu** [comment]
Just fyi - Storybook 7 just hit beta today.  If they happened to change something which could make your add-on now work, I will let you know... just not sure when I will get the chance to try it out just yet.

**Comment by mikestu** [comment]
I tried out the latest beta.1 version they have, and get similar results.  It looks  like they removed or changed a couple of things the add-on uses.  I don't have a repo with an example, but if it helps, here's a screenshot of the two errors:

![image](https://user-images.githubusercontent.com/3843165/206551504-ea75d295-54db-4f5f-99b0-fa998ed52b7a.png)

**Comment by fooloomanzoo** [comment]
https://github.com/nutboltu/storybook-addon-mock/pull/154

**Comment by mikestu** [comment]
Any update on this one? Thanks!

**Comment by danmaz74** [comment]
Still an issue...

**Comment by Cayan** [comment]
Looking forward for v7 support

**Comment by jonniebigodes** [comment]
Hey @nutboltu! I'm one of the Storybook maintainers. I focus primarily on the documentation and community outreach. I'm reaching out to you to get a bit of information regarding your future plans for the addon, and if you need any help from us with regards to any documentation changes or other changes that you might consider relevant, more even as it seems, there's interest in providing support for the upcoming 7.0 release, per the comments in this issue. To help you with this, we've prepared an abridged guide for the required changes [here](https://chromatic-ui.notion.site/Addon-migration-for-Storybook-7-002f2f627c1d470298ca94894ab8a040), and if you have any questions or issues, please reach out to us in the #prerelease channel in our [Discord Server](https://discord.gg/storybook). 

Looking forward to hearing from you.

Hope you have a fantastic day.

Stay safe

**Comment by mikestu** [comment]
For those that are waiting on this to support v7, another option is to use [Mock Service Worker](https://mswjs.io/) ([Storybook addon](https://storybook.js.org/addons/msw-storybook-addon)).  I have switched my project to it, and it went well so far. I won't close this though, as I'm sure others will need v7 here.

**Comment by nutboltu** [comment]
👋🏾 @everyone we have release version [4.0.0](https://github.com/nutboltu/storybook-addon-mock/releases/tag/4.0.0) which supports Storybook v7. A huge shout out to @fooloomanzoo  for taking the all hardship to make it happen.

**Comment by stoneqq11** [comment]
use v4.0.0 and storybook v7,when add mockAddonConfigs to preivew.js,thorw error:

**Comment by stoneqq11** [comment]
use v4.0.0 with storybook v7,when add empty mockAddonConfigs to preivew.js,throw error: 

```
mockAddonConfigs: {}
```

```
TypeError: undefined is not iterable (cannot read property Symbol(Symbol.iterator))
    at _iterableToArray (http://localhost:6006/vendors-node_modules_storybook_addon-essentials_dist_actions_preview_mjs-node_modules_storybo-3abf76.iframe.bundle.js:70736:76)
    at _toConsumableArray (http://localhost:6006/vendors-node_modules_storybook_addon-essentials_dist_actions_preview_mjs-node_modules_storybo-3abf76.iframe.bundle.js:70733:70)
    at withRoundTrip (http://localhost:6006/vendors-node_modules_storybook_addon-essentials_dist_actions_preview_mjs-node_modules_storybo-3abf76.iframe.bundle.js:70758:24)
    at hookified (http://localhost:6006/sb-preview/runtime.js:7:17241)
    at http://localhost:6006/sb-preview/runtime.js:34:1451
    at http://localhost:6006/sb-preview/runtime.js:34:1912
    at withOutline (http://localhost:6006/vendors-node_modules_storybook_addon-essentials_dist_actions_preview_mjs-node_modules_storybo-3abf76.iframe.bundle.js:1120:669)
    at hookified (http://localhost:6006/sb-preview/runtime.js:7:17241)
    at http://localhost:6006/sb-preview/runtime.js:34:1451
    at http://localhost:6006/sb-preview/runtime.js:34:1912
```

if add some config,then throw another error:

```
mockAddonConfigs: {
  refreshStoryOnUpdate: true
}
```

```
Uncaught Error: Rendered fewer hooks than expected. This may be caused by an accidental early return statement.
    at renderWithHooks (chunk-PUVLCKJJ.js:69:1937)
    at updateFunctionComponent (chunk-PUVLCKJJ.js:69:48263)
    at beginWork (chunk-PUVLCKJJ.js:71:21338)
    at HTMLUnknownElement.callCallback2 (chunk-PUVLCKJJ.js:2:1921)
    at Object.invokeGuardedCallbackDev (chunk-PUVLCKJJ.js:2:2445)
    at invokeGuardedCallback (chunk-PUVLCKJJ.js:2:3703)
    at beginWork$1 (chunk-PUVLCKJJ.js:90:39484)
    at performUnitOfWork (chunk-PUVLCKJJ.js:90:21018)
    at workLoopSync (chunk-PUVLCKJJ.js:90:20642)
    at performSyncWorkOnRoot (chunk-PUVLCKJJ.js:90:13607)
```

**Comment by nagman** [comment]
Has anyone succeeded in implementing this package with storybook `^7.5.3`?
I have the panel with controls to set delay, response, and so forth.
I don't have any error.

But when submitting my form that makes a fetch to `/api/contact`, the addon doesn't seem to be triggered at all.
Instead, I instantly get this error from `faker.js:119`:
```
http://localhost:6006/api/contact 404 (Not Found)

**Comment by terrymun** [comment]
@stoneqq11 Ran into the same issue, but it seems like you just need to add an empty array, i.e.:

```ts
mockAddonConfigs: {
  globalMockData: [],
  refreshStoryOnUpdate: true,
}
```
